nautobot/extras/datasources/registry.py

```diff
@@ -29,16 +29,13 @@ def refresh_datasource_content(model_name, record, request, job_result, delete=F
     Args:
         model_name (str): Identifier of the datasource owner, such as "extras.gitrepository"
         record (models.Model): Datasource model instance, such as a GitRepository record
-        request (HttpRequest): Initiating request for this refresh, optional, used for change logging if provided
+        user (User): Initiating user for this refresh, optional, used for change logging if provided
         job_result (JobResult): Passed through to the callback functions to use with logging their actions.
         delete (bool): True if the record is being deleted; False if it is being created/updated.
     """
     job_result.log(f"Refreshing data provided by {record}...", level_choice=LogLevelChoices.LOG_INFO)
-    job_result.save()
-    if request:
-        change_context = JobChangeContext(user=request.user)
-        # TODO(jathan): Review how we can eliminate these deliberate `set_status()` and `save()`
-        # calls,moving them to Celery-friendly primitives instead.
+    if user:
+        change_context = JobChangeContext(user=user)
         with change_logging(change_context):
             for entry in get_datasource_contents(model_name):
```

Comment:
Should be moved to a helper method so we don't have to maintain 2 copies of the same code